src/data/list/chain.lean

```diff
@@ -72,6 +72,29 @@ theorem chain_map_of_chain {S : β → β → Prop} (f : α → β)
   (p : chain R a l) : chain S (f a) (map f l) :=
 (chain_map f).2 $ p.imp H
 
+theorem chain_pmap_of_chain {S : β → β → Prop} {p : α → Prop}
+  (f : Π a, p a → β)
+  (H : ∀ a b ha hb, R a b → S (f a ha) (f b hb))
+  {a : α} (l : list α)
+  (hl₁ : chain R a l) (ha : p a) (hl₂ : ∀ a ∈ l, p a) :
+  chain S (f a ha) (list.pmap f l hl₂) :=
+begin
+  induction l generalizing a,
+  { simp },
+  { simp [H _ _ _ _ (rel_of_chain_cons hl₁), l_ih _ (chain_of_chain_cons hl₁)] }
```

Comment:
You're referring to the auto generated name `l_ih` here, could you name it explicitly in the induction call? Same below

Reply:
Thanks for the suggestion, done!